### PR TITLE
Add ProgramStatusTrigger and ScheduleDetail relevant changes to docs

### PR DESCRIPTION
**Pull request by sameetandpotatoes**
https://builds.cask.co/browse/CDAP-DQB380-1

- 4.3 documentation for ProgramStatusTriggers, ScheduleDetail changes

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
Closing the PR since the JIRA is moved.

File: cdap-docs/developer-manual/source/building-blocks/schedules.rst [hunk 1]
```diff
@@ -49,6 +51,22 @@ the same workflow is executing at that time. Note that:
...
+to create a specific *Trigger* to be used as follows::
+
+  schedule(
+    buildSchedule("Workflow1AndWorkflow2CompletedSchedule", ProgramType.WORKFLOW, "TriggeredWorkflow")
+      .triggerOn(getTriggerFactory().and(getTriggerFactory().onProgramStatus(ProgramType.WORKFLOW, "Workflow1",
+                                                                             ProgramStatus.COMPLETED),
+                                         getTriggerFactory().onProgramStatus(ProgramType.WORKFLOW, "Workflow2",
+                                                                             ProgramStatus.COMPLETED)));
+
+This schedule uses an and trigger that can only be satisfied when both Workflow1
```
Comment (anwar6953) [inline cdap-docs/developer-manual/source/building-blocks/schedules.rst]: I feel like the "and" should be wrapped in with a ` or quotes or something special, otherwise it can be hard (or confusing) to read the sentence.

Comment (ecfm) [reply cdap-docs/developer-manual/source/building-blocks/schedules.rst]: or uppercase `AND`?

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 2]
```diff
@@ -821,8 +821,7 @@ The request body is a JSON object specifying the details of the schedule to be c
         "type": "<trigger type>",
         ...
       },
-      "timeoutMillis": <timeout in milliseconds>,
-      "scheduleStatus": "<status of the schedule, SUSPENDED or SCHEDULED>"
```
Comment (anwar6953) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: why was this removed? Isn't it a part of the response JSON?

Comment (ecfm) [reply cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: The title is request body, so I think it should not include the status? Accidentally removed `timeoutMillis`, will put it back

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 3]
```diff
@@ -857,11 +857,30 @@ or a program status trigger::
       "type": "PROGRAM_STATUS"
     }
 
-or an `and` trigger::
+or an `and` trigger, where "triggers" can be a non-empty list of any type of triggers::
```
Comment (anwar6953) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: I think "can be" should be replaced with "is"

File: cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst [hunk 4]
```diff
@@ -857,11 +857,30 @@ or a program status trigger::
...
+        {
+          "type": "<trigger type>",
+          ...
+        },
+        ...
+      ],
+      "type": "AND"
+    }
+
+or an `or` trigger, where "triggers" can be a non-empty list of any type of triggers::
```
Comment (anwar6953) [inline cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst]: I think "can be" should be replaced with "is"

**Review by anwar6953 — APPROVED** [review]

**Comment by anwar6953** [comment]
LGTM. Just be sure to review the built docs and see how they look before merging!

File: cdap-docs/developer-manual/source/building-blocks/schedules.rst [hunk 5]
```diff
@@ -147,7 +167,7 @@ Triggers
 
 A trigger can be based on time or data availability, These are the available trigger types:
 
-- ``.triggerByTime(String cronExpression)``: This triggers every time the cron expression
+- ``.triggerByTime(String cronExpression)``: This is triggered every time the cron expression
```
Comment (sagarkapare) [inline cdap-docs/developer-manual/source/building-blocks/schedules.rst]: @maochf I don't think we should have method syntax over here since its already covered by javadoc?
We can simply say  - Trigger types are - Time based, Partition based, Program status based and explain what each one means.

File: cdap-docs/developer-manual/source/building-blocks/schedules.rst [hunk 6]
```diff
@@ -167,37 +166,19 @@ Triggers
...
-  - ``TriggerFactory#and(Trigger... triggers)``: This is triggered when any of the given
-    triggers in the parameter is/are satisfied.
+- ``partition``: This is triggered by new partitions in the named dataset (it must be a
+  ``PartitionedFileSet`` or a ``TimePartitionedFileSet``). While a single notification for
+  new partitions in the dataset suffices to to create a job for the schedule, it will
+  remain in *pending trigger* state until enough partitions have arrived to exceed
+  the ``numPartitions`` parameter.
+- ``program status``: This is triggered when a given program in the specific namespace,
+  application, and application version transitions to any one of the given program statuses.
+- ``and``: This is triggered when all the triggers contained in it are satisfied.
```
Comment (sagarkapare) [inline cdap-docs/developer-manual/source/building-blocks/schedules.rst:180]: `and` is not the type of trigger right but multiple triggers can be combined using `and`, `or` constructs.  Maybe instead of putting it in the list describe in separate paragraph what `and` and `or` does?

**Comment by sagarkapare** [comment]
@maochf one comment, otherwise LGTM 👍

File: cdap-docs/developer-manual/source/building-blocks/schedules.rst (hunk 6 above)
Comment (ecfm) [reply cdap-docs/developer-manual/source/building-blocks/schedules.rst:180]: however, in the `lifecycle.rst`, `and`/`or` are one of the trigger types that can be used as the value of "type" property in the trigger json object, and they are actually treated also as a type of trigger. `and`/`or` is a single trigger that can contain multiple triggers, which can also include other `and`/`or` triggers
